lms/djangoapps/courseware/access.py

```diff
@@ -507,14 +536,14 @@ def can_load():
         students to see modules.  If not, views should check the course, so we
         don't have to hit the enrollments table on every module load.
         """
-        return (
-            not descriptor.visible_to_staff_only
+        return _has_staff_access_to_descriptor(user, descriptor, course_key) or (
+            _visible_to_nonstaff_users(descriptor)
             and _has_group_access(descriptor, user, course_key)
             and (
-                'detached' in descriptor._class_tags  # pylint: disable=protected-access
+                _has_detached_class_tag(descriptor)
                 or _can_access_descriptor_with_start_date(user, descriptor, course_key)
             )
-        ) or _has_staff_access_to_descriptor(user, descriptor, course_key)
+        )
```

Comment:
As earlier: Please rearrange the logic here so the staff access check happens at the end (as it was before).  I believe the staff access check currently requires a sql database hit.